datafusion/datasource-parquet/src/opener.rs

```diff
@@ -122,11 +125,16 @@ impl FileOpener for ParquetOpener {
         );
         let enable_bloom_filter = self.enable_bloom_filter;
         let limit = self.limit;
+        let file_decryption_properties = self.file_decryption_properties.clone();
 
         Ok(Box::pin(async move {
-            let options = ArrowReaderOptions::new().with_page_index(enable_page_index);
+            let mut options = ArrowReaderOptions::new().with_page_index(enable_page_index);
+            if let Some(ref fd_val) = file_decryption_properties {
+                options = options.with_file_decryption_properties((**fd_val).clone());
+            }
```

Comment:
This is key for reading. Reads seem to go through ParquetOpener so we can hook in here. TODO: Discuss with Adam about hooking in file specific AAD.